### ISSUE TITLE
Unpublished stories are not shown until published with "Preview" token

### ISSUE THREAD

**Issue by marekrozmus** [storyblok/storyblok-react#242]
Please correct me if I am wrong.
I understand that when using the `Preview` token the API should also return unpublished stories.
But the problem is that I got 404 for the saved stories and they become available in the visual editor only after I make `Publish`.

As I understood that `Preview` token should give access to all stories and the `Public` only for the published ones?

**Comment by serrg** [comment]
via https://www.storyblok.com/docs/api/content-delivery :

> To make the API as explorable as possible, accounts have draft versions and published version of API tokens. To access the draft version of your content you can use the preview token, and for receiving published content you can use the public token. The preview token is able to also load the published content. To switch between those versions you can append the query parameter version=draft/published and using the appropriate token to perform a draft or published version call.

**Comment by fgiuliani** [comment]
Hi @marekrozmus!
Are you retrieving the "draft" version of your content from the code of your application?

```
const storyblokApi = getStoryblokApi();
let { data } = await storyblokApi.get(`cdn/stories/react`, {
  version: "draft",
});
```

**Comment by marekrozmus** [comment]
@fgiuliani  Yes, that is why I am asking. This is my code:
```javascript
let sbParams = {
    version: "draft", // or 'published'
    language: i18nextConfig.i18n.defaultLocale,
  };

  const storyblokApi = getStoryblokApi();
  let { data } = await storyblokApi.get(`cdn/stories/${slug}`, sbParams);
```

**Comment by fgiuliani** [comment]
@marekrozmus hmm, that is strange.
Have you saved your story at least once?

**Comment by marekrozmus** [comment]
@fgiuliani Yes. I am creating new story, add the Teaser component (the one from the tutorial) and save the story. Nothing is shown. When I publish it at least once then it starts working.

**Comment by fgiuliani** [comment]
@marekrozmus are you seeing the changes while you are creating the Teaser at first? I mean, you create a new story, you enter the visual editor, you add a Teaser and change the text of the property. Do you see the text being displayed while you type?

**Comment by marekrozmus** [comment]
@fgiuliani Here is a screen recording: https://youtu.be/uUTnM4SGV3c
Is it possible that it has something to do with usage of dynamic paths `[[...slug]].tsx`

**Comment by fgiuliani** [comment]
Thanks for the video @marekrozmus, it's really helpful.
Is there any way we can see the code of your `[[...slug]].tsx` file?

**Comment by marekrozmus** [comment]
@fgiuliani You can check with this repo: https://github.com/marekrozmus/storyblok-ts The same problem.
Also I get 404 when I unpublish the story. I am using the preview access token.

**Comment by fgiuliani** [comment]
Thanks for all the details @marekrozmus. I will contact the team and will come back to you with news.

**Comment by arorachakit** [comment]
Hey @marekrozmus , the issue here doesn't seem to be related to the token. 
As far as I can tell, the issue is that in getStaticPaths - where you fetch the links with ` let { data } = await storyblokApi.get("cdn/links/");` This `links` endpoint only fetches the published links (by default), and hence if you don't publish the story you won't be able to see the content. As there will be no link in the data, and getStaticProps won't be fetching the data for that link. 
Here is the documentation to links endpoint - https://www.storyblok.com/docs/api/content-delivery/v2#core-resources/links/links

To make this work, you will need to fetch the draft links as well, if you don't specify the version the links endpoint only fetches the published links by default. To add the version you can do this - 
`  let { data } = await storyblokApi.get("cdn/links/" ,{
    version: 'draft'
  });` 
  Inside your getStaticPaths. I think it should work after this :) Please let us know if you still face the isse.

**Comment by marekrozmus** [comment]
Hi @arorachakit. You are right it works now 😄 .
I was using the [tutorial](https://www.storyblok.com/tp/add-a-headless-cms-to-next-js-in-5-minutes) and and following code mislead me:
```javascript
let sbParams = {
    version: "draft", // or 'published'
};
``` 
I thought that this part of code and token are enough.

Other thing is that the link to the `links` API doesn't say anything about query params it supports or I just can't find it. It says about endpoint only. Other endpoints have a supported query params table (or at least some of them ;))

Thank you for the answer 🥳  🚀

**Comment by arorachakit** [comment]
You can find the query parameters just below that. 😄
https://www.storyblok.com/docs/api/content-delivery/v2#core-resources/links/retrieve-multiple-links

**Comment by fgiuliani** [comment]
Hi @coreybruyere what do you think it's missing on the tutorial?

cc @arorachakit

**Comment by arorachakit** [comment]
Hey @coreybruyere - we made an update a few months back to fix that. 
You can find it here - https://www.storyblok.com/tp/render-storyblok-stories-dynamically-in-next-js#dynamic-route-generation-with-getstaticpaths
